backend/views/syson-diagram-statetransition-view/src/main/java/org/eclipse/syson/diagram/statetransition/view/STVDescriptionNameGenerator.java

```diff
@@ -21,7 +24,29 @@
  */
 public class STVDescriptionNameGenerator extends DescriptionNameGenerator {
 
+    public static final String ACTION = "action";
+
+    public static final String ACTIONS = "actions";
+    
     public STVDescriptionNameGenerator() {
         super("STV");
     }
+    
+    @Override
+    public String getCompartmentName(EClass eClass, EReference eReference) {
+        if (new StateTransitionActionToolProvider(eReference).isHandledAction()) {
+            return this.getCompartmentName(this.getDiagramPrefix(), eClass.getName() + SPACE + ACTIONS);
+        } else {
+            return this.getCompartmentName(this.getDiagramPrefix(), eClass.getName() + SPACE + eReference.getName());
+        }
+    }
+    
+    @Override
+    public String getCompartmentItemName(EClass eClass, EReference eReference) {
+        if (new StateTransitionActionToolProvider(eReference).isHandledAction()) {
+            return this.getCompartmentName(this.getDiagramPrefix(), eClass.getName() + SPACE + ACTION);
```

Comment:
Sorry, I didn't see that before.
Are you sure that it should be `this. getCompartmentName (...)` here? I would have used `this. getCompartmentItemName (...)` but may be there is a good reason.